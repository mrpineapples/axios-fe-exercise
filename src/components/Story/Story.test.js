import { render, screen } from "@testing-library/react";
import { Story } from ".";
import { resizeWindow } from "../../../test/utils";

const data = {
    id: "57a186a3-7547-45bf-852a-83019849d8d5",
    headline: "Coronavirus dashboard",
    published_date: "2021-08-29T19:07:12.354000Z",
    permalink:
        "https://www.axios.com/coronavirus-latest-news-quick-highlights-57a186a3-7547-45bf-852a-83019849d8d5.html",
    primary_image: {
        alt_text: "Some alt text",
        base_image_url:
            "https://images.axios.com/pzV7Tj0m2AhS4XN8XsvWP2ot8Vo=/fit-in/1366x1366/2021/08/27/1630088719253.jpg",
    },
    topics: [
        {
            id: "651f9439-6eb5-4d4f-a6fc-2587ecf17eeb",
            slug: "politics-policy",
            name: "Politics & Policy",
            description:
                "President Biden has made the COVID-19 crisis and a post-Trump return to national unity and traditional democratic ideals his top priorities. From vaccinations to stimulus to schools, Biden is seeking bipartisan compromise while showing a willingness to use executive authority and bare Democratic majorities in the U.S. House and Senate to implement his policies. Republican leaders are navigating deep party divisions over if and how to move beyond former President Trump.",
            nav_position: 10,
            show_in_nav: true,
            show_in_eden: true,
        },
    ],
};

const mockStory = {
    key: data.id,
    altText: data.altText,
    headline: data.headline,
    topic: data.topics[0].name,
    permaLink: data.permalink,
    publishedDate: data.published_date,
    imageURL: data.primary_image.base_image_url,
};

describe("<Story />", () => {
    test("Should render correctly", () => {
        const { container } = render(<Story {...mockStory} />);
        expect(container).toMatchSnapshot();
    });

    describe("On desktop", () => {
        test("Should render deep link", () => {
            render(<Story {...mockStory} />);
            expect(screen.getByTestId("deep-link")).toBeInTheDocument();
        });

        test("Should render topic as a separate element", () => {
            render(<Story {...mockStory} />);
            expect(screen.getByTestId("topic-desktop")).toBeInTheDocument();
        });
    });

    describe("On mobile", () => {
        test("Should not render deep link", () => {
            resizeWindow(375, 812);
            render(<Story {...mockStory} />);
            expect(screen.queryByTestId("deep-link")).toBeNull();
        });

        test("Should render topic alongside date", () => {
            resizeWindow(375, 812);
            render(<Story {...mockStory} />);
            expect(screen.getByTestId("topic-mobile")).toBeInTheDocument();
        });
    });
});
