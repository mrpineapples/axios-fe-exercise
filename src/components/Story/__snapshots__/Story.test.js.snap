// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Story /> Should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: inherit;
}

.c0 .story-img {
  width: 100%;
  height: 135px;
}

.c0 .headline {
  font-size: 18px;
  margin: 16px 0 12px 0;
}

.c0 .topic--desktop {
  font-size: 14px;
  color: #ab7d36;
  margin: 20px 0 8px 0;
}

.c0 .story-meta {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #656568;
  font-size: 12px;
}

.c0 .story-meta .topic:before {
  content: " - ";
}

.c0 .story-meta .link {
  font-size: 18px;
  margin-left: auto;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #2257da;
}

@media (min-width:980px) {
  .c0 .story-img {
    height: 205px;
  }

  .c0 .story-meta {
    font-size: 14px;
  }
}

<div>
  <div
    class="c0"
    data-testid="story-container"
  >
    <a
      href="https://www.axios.com/coronavirus-latest-news-quick-highlights-57a186a3-7547-45bf-852a-83019849d8d5.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        class="story-img"
        src="https://images.axios.com/pzV7Tj0m2AhS4XN8XsvWP2ot8Vo=/fit-in/1366x1366/2021/08/27/1630088719253.jpg"
      />
    </a>
    <div
      class="info"
    >
      <p
        class="topic topic--desktop"
        data-testid="topic-desktop"
      >
        Politics & Policy
      </p>
      <p
        class="headline"
      >
        Coronavirus dashboard
      </p>
      <div
        class="story-meta"
      >
        <time
          class="date"
          datetime="2021-08-29T19:07:12.354000Z"
        >
          August 29, 2021
        </time>
        <a
          class="link"
          data-testid="deep-link"
          href="https://www.axios.com/coronavirus-latest-news-quick-highlights-57a186a3-7547-45bf-852a-83019849d8d5.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          Go deeper â†’
        </a>
      </div>
    </div>
  </div>
</div>
`;
